unsupported/Eigen/CXX11/src/Tensor/TensorReductionGpu.h

```diff
@@ -289,6 +339,12 @@ struct FullReductionLauncher {
   }
 };
 
+#if defined(EIGEN_HIPCC)
+namespace {
+  std::mutex __eigen_reduction_hip_mutex;
```

Comment:
this is not needed anymore. it was added to support tensorflow but since we switched to `rocPRIM` it's not needed anymore.